server/sonar-server/src/main/java/org/sonar/server/computation/task/projectanalysis/scm/ScmInfoRepositoryImpl.java

```diff
@@ -87,6 +103,24 @@ private static ScmInfo getScmInfoFromReport(Component file, ScannerReport.Change
     return new ReportScmInfo(changesets);
   }
 
+  private ScmInfo generatedScmInfo(Component file) {
+    ScmInfo dbInfo = scmInfoDbLoader.getScmInfoFromDb(file);
+    if (dbInfo == NoScmInfo.INSTANCE) {
+      Set<Integer> newOrChangedLines = sourceLinesDiff.getNewOrChangedLines(file);
+      if (newOrChangedLines.isEmpty()) {
+        return NoScmInfo.INSTANCE;
+      }
+      return new GeneratedScmInfo(analysisMetadata.getAnalysisDate(), newOrChangedLines);
+    } else {
+      // TODO merge with DB
```

Comment:
![Code Smell](https://sonarsource.github.io/sonar-github/code_smell.png 'Code Smell') Code Smell: Complete the task associated to this TODO comment. ([squid:S1135](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS1135))

[See it in SonarQube](https://next.sonarqube.com/sonarqube/project/issues?id=org.sonarsource.sonarqube%3Asonarqube&branch=feature%2Fdm%2Fno_scm_info&issues=AWD56Pl1EFfjxkQcPmlU&open=AWD56Pl1EFfjxkQcPmlU)

[ISSUE_KEY]: AWD56Pl1EFfjxkQcPmlU